src/toil/test/batchSystems/batchSystemTest.py

```diff
@@ -195,6 +196,64 @@ def _waitForJobsToStart(self, numJobs):
                 runningIDs = self.batchSystem.getRunningBatchJobIDs().keys()
             return runningIDs
 
+    class AbstractBatchSystemJobTest(ToilTest):
+        """
+        A base test case with generic tests that every batch system should pass
```

Comment:
`An abstract base class for batch system tests that use a full Toil workflow rather than using the batch system directly.`